device-registry/src/main/scala/org/genivi/sota/device_registry/DeviceRepository.scala

```diff
@@ -95,45 +93,28 @@ object Devices {
       .flatMap(_.
         fold[DBIO[Device]](DBIO.failed(Errors.MissingDevice))(DBIO.successful))
 
-  def findByDeviceName(ns: Namespace, deviceName: DeviceName)
-                      (implicit ec: ExecutionContext): DBIO[Device] =
-    devices
-      .filter(d => d.namespace === ns && d.deviceName === deviceName)
-      .result
-      .headOption
-      .flatMap(_.
-        fold[DBIO[Device]](DBIO.failed(Errors.MissingDevice))(DBIO.successful))
-
   def findByDeviceId(ns: Namespace, deviceId: DeviceId)
-                    (implicit ec: ExecutionContext): DBIO[Device] =
+                    (implicit ec: ExecutionContext): DBIO[Seq[Device]] =
     devices
       .filter(d => d.namespace === ns && d.deviceId === deviceId)
       .result
-      .headOption
-      .flatMap(_.
-        fold[DBIO[Device]](DBIO.failed(Errors.MissingDevice))(DBIO.successful))
 
   def search(ns: Namespace, re: String Refined Regex): DBIO[Seq[Device]] =
     devices
       .filter(d => d.namespace === ns && regex(d.deviceName, re))
       .result
 
   def update(ns: Namespace, id: Id, device: DeviceT)
-            (implicit ec: ExecutionContext): DBIO[Unit] =
-    device.deviceId match {
-      case Some(deviceId) => for {
-        _ <- exists(ns, id)
-        _ <- findByDeviceId(ns, deviceId).asTry.flatMap { // negate result of action:
-          case Success(_) => DBIO.failed(Errors.ConflictingDevice)
-          case Failure(_) => DBIO.successful(())
-        }
-        _ <- devices.insertOrUpdate(Device(ns, id, device.deviceName, device.deviceId, device.deviceType))
-      } yield ()
-      case None => for {
-        _ <- exists(ns, id)
-        _ <- devices.insertOrUpdate(Device(ns, id, device.deviceName, device.deviceId, device.deviceType))
-      } yield ()
-    }
+            (implicit ec: ExecutionContext): DBIO[Unit] = {
+
+    val dbIO = for {
+      _ <- exists(ns, id)
+      _ <- notConflicts(ns, device.deviceName, device.deviceId)
+      _ <- devices.update(Device(ns, id, device.deviceName, device.deviceId, device.deviceType))
+    } yield ()
+
+    dbIO.transactionally
```

Comment:
@simao if you mean this, i took care of it no?
all tests were green, dunno why it would show them as red